src/main/java/hudson/plugins/ec2/EC2RetentionStrategy.java

```diff
@@ -110,14 +110,23 @@ private long internalCheck(EC2Computer computer) {
             InstanceState state;
 
             try {
+                state = computer.getState(); //Get State before Uptime because getState will refresh the cached EC2 info
                 uptime = computer.getUptime();
-                state  = computer.getState();
```

Comment:
I'm not getting this bit, unless `getUptime` somehow the instance changes its launch time. This should not change anything. The change is still harmless though.

Reply:
Yes, that's exactly right - getState() causes the stale EC2 instance "description" (including launch time) to be updated from AWS.

The instance description is cached unless explicitly flushed by a call to getState() or get updateInstanceDescription().  There are several cases where the cached value may not be accurate, of particular importance is when an instance is being launched.  When the instance is launched, a start is requested, but the instance description isn't flushed until we try to connect via ssh or WinRM (which is after the launch timeout).

What ends up happening as a result is that if the EC2RetentionStrategy happens between the time that a instance is initially launched and the launch timeout, it may appear to be idle based on the stale previous idle time and have the idle timeout triggered.  By switching the order so getState() refreshes the instance description this situation is avoided.